### PR TITLE
KAFKA-18278: Correct name and description for run-gradle step

### PR DESCRIPTION
**Pull request by LoganZhuZzz**
*The name and description for the `run-gradle` step were updated to make them clearer and more consistent with the overall workflow. The updated names now better describe the purpose of the step.*

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: .github/actions/run-gradle/action.yml [hunk 1]
```diff
@@ -16,8 +16,8 @@
 # under the License.
 #
 ---
-name: "Gradle Setup"
-description: "Setup Java and Gradle"
+name: "Gradle Test Setup"
```
Comment (mumrah) [inline .github/actions/run-gradle/action.yml]: This was a mis-copied name from our setup-gradle action. It should just be "Run Tests with Gradle" or similar.

**Review by mumrah — CHANGES_REQUESTED** [review]

File: .github/actions/run-gradle/action.yml [hunk 2]
```diff
@@ -16,8 +16,8 @@
 # under the License.
 #
 ---
-name: "Gradle Setup"
-description: "Setup Java and Gradle"
+name: "Run Gradle Tests"
```
Comment (chia7712) [inline .github/actions/run-gradle/action.yml]: Could you please use "Run Tests with Gradle"? It makes more sense to me.

Comment (LoganZhuZzz) [reply .github/actions/run-gradle/action.yml]: It makes more sense. I’ve made the change accordingly.

**Review by mumrah — APPROVED** [review]
LGTM!

**Pull request merged**